src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1862,8 +1865,8 @@ GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
 	List *joinRestrictClauseList = get_all_actual_clauses(joinRestrictInfo);
 	if (ContainsFalseClause(joinRestrictClauseList))
 	{
-		/* found WHERE false, no need  to continue */
-		return copyObject((List *) joinRestrictClauseList);
+		/* found WHERE false, no need  to continue, we just return a false clause */
+		return GetFalseClause(joinRestrictClauseList);
```

Comment:
We can possible generate a false clause from scratch as well

Reply:
makes sense